packages/onboard-helpers/src/hooks/useOnboardHelpers.ts

```diff
@@ -0,0 +1,356 @@
+        const { walletChainId } = getWallet(parsedWallet)
+
+        // check wallet sync again
+        if (parsedWallet && walletChainId && +walletChainId !== chainId) {
+          parsedWallet = await handleReconnectWallet(parsedWallet, chainId)
+        }
+
+        await updateApi(chainId, useWallet ? parsedWallet : null)
+        updateConnectState('success', '')
+      } catch (error) {
+        console.error(error)
+        updateConnectState('failure', CONNECT_STAGE.CONNECT_API)
+      }
+    },
+    [wallet, updateApi, updateConnectState, handleReconnectWallet]
+  )
+
+  const handleConnectWallet = useCallback(
+    async (walletName: ConnectOptions[CONNECT_STAGE.CONNECT_WALLET]) => {
+      let walletState
+
+      if (!walletName) {
+        walletState = (await connect())?.[0]
+      } else {
+        // If found label in localstorage, after 30s if not connected, reconnect with modal
+        const walletStatesPromise = new Promise<WalletState[] | null>(async (resolve, reject) => {
+          try {
+            const walletStates = await Promise.race([
+              connect({ autoSelect: { label: walletName, disableModals: true } }),
+              new Promise<never>((_, reject) => setTimeout(() => reject(new Error('timeout connect wallet')), 3000)),
+            ])
+            resolve(walletStates)
+          } catch (error) {
+            reject(error)
+          }
+        })
+
+        try {
+          const walletStates = await walletStatesPromise
+          if (!walletStates || walletStates?.length === 0) throw new Error('unable to connect')
+          walletState = walletStates[0]
+        } catch (error) {
+          // if failed to get walletState due to timeout, show connect modal.
+          updateStorageWalletName('')
+          ;[walletState] = await connect()
+        }
+      }
+
+      try {
+        if (!walletState) throw new Error('No wallet found')
+
+        updateStorageWalletName(walletState.label)
+        const { walletChainId, walletChainIdHex } = getWallet(walletState)
+
+        if (walletChainId && walletChainIdHex) {
+          let parsedChainId = +walletChainId
+
+          // if route and wallet chainId does not match, update wallet's chainId
+          if (+walletChainId !== routeChainId) {
+            const success = await setChain({ chainId: ethers.toQuantity(routeChainId) })
+            if (!success) throw new Error('reject network switch')
+            parsedChainId = routeChainId
+          }
+
+          // confirm wallet's chainId is updated
+          const { isNetworkSwitched } = await _isNetworkSwitched(routeChainId)
+
+          if (!isNetworkSwitched) {
+            const { networkId, isActiveNetwork } = getNetwork(parsedChainId)
+
+            if (networkId && isActiveNetwork) {
+              navigate(`${routeLocalePathname}/${networkId}/${routeFullPathname}`)
+              updateConnectApi([parsedChainId, true])
+            } else {
```

Comment:
Try not to use else but use return where possible to reduce code nesting
```suggestion
              return
            }
```